doc/src/modules/solvers/solveset.rst

```diff
@@ -1,12 +1,463 @@
+   given equation in complex and real domain respectively.
+
+ * If the given function (equation) is a product of two or more functions,
+   like say `f = g*h`, then the solution to the given equation is the Union
+   of the solution of the equations `g = 0` and `h = 0`, if and only if both
+   `g` and `h` are finite for a finite input. So, the solution is build up
+   recursively.
+
+ * The function class is now checked if it's Trigonometric or Hyperbolic, then
+   the function ``\_solve\_real\_trig`` is called, which solves it by converting
+   it in terms of ``exp`` complex exponential form.
+
+ * The function is now checked if there is any instance of Piecewise
+   expression, if it is, then it's converted to explict expression and
+   set pairs and then solved recursively.
+
+ * The respective solvers now tries to invert the equation using the routines
+   ``invert\_real`` and ``invert\_complex``.
+
+ * After the invert, the equations are checked for radical or Abs (Modulus),
+   then the method ``\_solve\_radical`` tries to simplify the radical, by
+   removing it using techniques like squarring, cubing etc, and ``\_solve\_abs``
+   solves nested Modulus by considering the positive and negative variants,
+   iteratively.
+
+ * If none of the above method is successful, then method of polynomial is
+   used as follows:
+
+   - ``\_solve\_as\_rational`` is called, it's third argument is the
+     ``solveset\_solver`` which can either be ``solveset_real`` or
+     ``solveset\_complex`` based on these respective poly solvers
+     ``\_solve\_as\_poly\_real`` and ``\_solve\_as\_poly\_complex`` is called.
+
+   - The underlying method ``\_solve\_as\_poly`` solves the equation using
+     polynomial techniques if it already is a polynomial equation or, with
+     a change of variables, can be made so.
+
+ * The Final solution set obtained is taken intersection with the input
+   domain, and the resultant solution is returned.
+
+.. Remember to change the above part when the new solver is implemented.
+
+
+How do we manipulate and return an infinite solution?
+-----------------------------------------------------
+
+ * In Real Domain, we use our ``ImageSet`` class in the sets module to
+   return infinite solutions. ``ImageSet`` is an Image of a set under
+   a mathematical function. For example, to represent the solution
+   of the equation `sin(x) == 0`, we can use the ImageSet as:
+
+   ``ImageSet(Lambda(n, 2*n*pi), Integers())``
+
+   Where n is a dummy variable. It is basically the image of the
+   Integers set under the function `2*n*pi`.
+
+ * In Complex Domain, we use Complex Sets, which is implemented as
+   ``ComplexPlane`` class in the sets module, to represent infinite
+   solution in the argand plane. For example to represent the solution
+   of the equation `|z| == 1`, which is a unit circle, we can use
+   the ComplexPlane as:
+
+   ``ComplexPlane(FiniteSet(1)*Interval(0, 2*pi), polar=True)``
```

Comment:
I would do these as doctests to ensure that they are and stay correct, at least syntactically. Plus it will render nicer.